backend/models.py

```diff
@@ -0,0 +1,159 @@
+import enum
+from datetime import datetime
+from typing import List, Union, Optional, NamedTuple
+
+from sqlalchemy import (
+    select, Enum, DateTime,
+    Boolean, Column, ForeignKey, 
+    Integer, String, Float)
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm.session import Session
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.sql.schema import UniqueConstraint
+
+from .database import Base
+
+class Value(Base):
+    __abstract__ = True
+    
+    id = Column(Integer, primary_key=True, index=True)
+    @declared_attr
+    def entity_id(cls):
+        return Column(Integer, ForeignKey('entities.id'))
+
+    @declared_attr
+    def attribute_id(cls):
+        return Column(Integer, ForeignKey('attributes.id'))
+
+    @declared_attr
+    def entity(cls):
+        return relationship('Entity')
+
+class ValueBool(Value):
+    __tablename__ = 'values_bool'
+    value = Column(Boolean)
+
+
+class ValueInt(Value):
+    __tablename__ = 'values_int'
+    value = Column(Integer)
+
+
+class ValueFloat(Value):
+    __tablename__ = 'values_float'
+    value = Column(Float)
+
+
+class ValueForeignKey(Value):
+    __tablename__ = 'values_fk'
+    value = Column(Integer)
+
+
+class ValueStr(Value):
+    __tablename__ = 'values_str'
+    value = Column(String)
+
+
+class ValueDatetime(Value):
+    __tablename__ = 'values_datetime'
+    value = Column(DateTime)
+
+
+class Mapping(NamedTuple):
+    model: Value
+    converter: type
+
+
+class AttrType(enum.Enum):
+    STR = Mapping(ValueStr, str)
+    BOOL = Mapping(ValueBool, bool)
+    INT = Mapping(ValueInt, int)
+    FLOAT = Mapping(ValueFloat, float)
+    FK = Mapping(ValueForeignKey, int)
+    DT = Mapping(ValueDatetime, datetime.fromtimestamp)
+
+
+class BoundFK(Base):
+    __tablename__ = 'bound_foreign_keys'
+
+    id = Column(Integer, primary_key=True, index=True)
+    attr_def_id = Column(Integer, ForeignKey('attr_definitions.id'))
+    schema_id = Column(Integer, ForeignKey('schemas.id'))
```

Comment:
Perhaps we should change `BoundFK.schema` name to `BoundFK.bound_schema` to make it more clear

Reply:
Oh, right. Sorry, I completely forgot about that. I don't think it's necessary to rename the field; it might be sufficient to add a `doc` or `comment` parameter to it.